File: celula.cpp
/**Archivo de descripción del objeto celula
 * 
 * 
*/

#include <cstdio>
using namespace std;

class celula
{
private:
    short int _estado;
    int _periodos;
    //objeto de interfaz
public:
    //constantes estaticas
    const static short int estado_celula_sana = 0;
    const static short int estado_celula_infentada_A = 1;
    const static short int estado_celula_infectada_B = 2;
    const static short int estado_celula_muerta = 3;

    celula(){
        this->_estado = celula::estado_celula_sana;
    }
    celula(short int estado){
        this->_estado = estado;
    }

    short int get_estado(){
        return this->_estado;
    }

    void set_estado(short int nuevo_estado){
        this->_estado = nuevo_estado;
    }

     int get_periodo(){
        return this->_periodos;
    }

    void set_periodos(int t){
        this->_periodos = t;
    }

    void paso_periodo(){
        this->_periodos --;
    }

};

File: Celula.java
package automatacelular;

public class Celula {
    public static final int estado_celula_sana = 0;
    public static final int estado_celula_infentada_A = 1;
    public static final int estado_celula_infentada_B = 2; // Corregido el nombre de la variable
    public static final int estado_celula_muerta = 3;

    private int estado;
    private int estadoFuturo;
    private int periodosRestantes;

    public Celula() {
        estado = estado_celula_sana;
        estadoFuturo = estado_celula_sana;
        periodosRestantes = 0;
    }

    public int getEstado() {
        return estado;
    }

    public void setEstado(int estado) {
        this.estado = estado;
    }

    public int getEstadoFuturo() {
        return estadoFuturo;
    }

    public void setEstadoFuturo(int estadoFuturo) {
        this.estadoFuturo = estadoFuturo;
    }

    public int getPeriodo() {
        return periodosRestantes;
    }

    public void setPeriodos(int periodos) {
        this.periodosRestantes = periodos;
    }

    public void pasoPeriodo() {
        this.periodosRestantes--;
    }
}

